app/src/main/java/com/example/sharewhatyoucanproject/login/LoginActivity.kt

```diff
@@ -0,0 +1,110 @@
+        progressDialog.setTitle("Please Wait")
+        supportActionBar?.hide()
+
+        loginViewModel = ViewModelProvider(
+            this,
+            LoginViewModelFactory(),
+        )[LoginViewModel::class.java]
+
+        val type = intent.getIntExtra("type", 0)
+
+        val typevar1 = UserType.values()[type]
+
+        loginViewModel.type = typevar1
+
+        loginViewModel.checkResult.observe(this) { result ->
+            result?.let {
+                progressDialog.dismiss()
+                // this.showToast(result)
+                // TODO show toast
+                loginViewModel.checkResult.value = null
+            }
+        }
+
+        loginViewModel.loginResult.observe(this) { loginResult ->
+            loginResult?.let {
+                progressDialog.dismiss()
+                if (loginResult == "success") {
+                    editSharedPreferencesSelector(this, loginViewModel.type)
+                    val intent = Intent(this@LoginActivity, DrawerActivity::class.java)
+                    startActivity(intent)
+                    finishAffinity()
+                } else {
+                    this.showToast(loginResult)
+                }
+                loginViewModel.loginResult.value = null
+            }
+        }
+
+        loginViewModel.createUserResult.observe(this) { createUserResult ->
+            createUserResult?.let {
+                if (createUserResult == "success") {
+                    this.showToast("Created")
+                    loginViewModel.saveUserData()
+                } else {
+                    progressDialog.dismiss()
+                    this.showToast(createUserResult)
+                }
+                loginViewModel.createUserResult.value = null
+            }
+        }
+
+        loginViewModel.saveUserResult.observe(this) { saveUserResult ->
+            saveUserResult?.let {
+                progressDialog.dismiss()
+                if (saveUserResult == "success") {
+                    editSharedPreferencesSelector(this, loginViewModel.type)
+                    val intent = if (loginViewModel.type == UserType.DONER) {
+                        Intent(this@LoginActivity, SelectUserActivity::class.java)
+                    } else {
+                        Intent(this, DrawerActivity::class.java)
+                    }
+                    startActivity(intent)
+                    finishAffinity()
+                } else {
+                    this.showToast(saveUserResult)
+                }
+                loginViewModel.saveUserResult.value = null
+            }
+        }
+
+        if (loginViewModel.type == UserType.DONER) {
+            binding.mytv.text = "Donor"
+        } else if (loginViewModel.type == UserType.RECEIVER) {
+            binding.mytv.text = "Receiver"
+        }
+
+        binding.submitdevice.setOnClickListener {
+            val deviceId = Settings.Secure.getString(contentResolver, Settings.Secure.ANDROID_ID)
```

Comment:
Just curious, why use device id like this as the login identifier? It requires special permissions in later versions of android: https://developer.android.com/training/articles/user-data-ids. Using the same password for every account is a security issue. Is secure login required for this project?